src/Extremas.chpl

```diff
@@ -46,16 +46,31 @@ to increase the efficiency of the merge step.
     // instance, only pushes a value if
     // it is an encountered extreme
     proc push(val: (eltType, int)) {
-      const shouldAdd = if isMinReduction then val(0)<_data[0](0) else val(0)>_data[0](0);
-      const shouldAddEmpty = (numEmpty>0) && if isMinReduction then
-        _data[0](0)==max(eltType) else _data[0](0)==min(eltType);
 
-      if shouldAddEmpty {
-        _data[numEmpty+1] = val;
-        numEmpty-=1;
-      } else if shouldAdd {
-        _data[0] = val;
-        heapifyDown();
+      // Common case where _data is a heap
+      if !isSorted {
+        const shouldAdd = if isMinReduction then val(0)<_data[0](0) else val(0)>_data[0](0);
+        const shouldAddEmpty = (numEmpty>0) && if isMinReduction then
+          _data[0](0)==max(eltType) else _data[0](0)==min(eltType);
+
+        if shouldAddEmpty {
+          _data[numEmpty+1] = val;
+          numEmpty-=1;
+        } else if shouldAdd {
+          _data[0] = val;
+          heapifyDown();
+        }
+      } else {
+        // Uncommon for an accumulate to be called after combining, so does not
+        // need to be very fast/optimized.
```

Comment:
For Arkouda's use of this reduction, the accumulates after combine will always be with an identity value so nothing will actually be done. This could be `ifSorted then assert(val == if isMinReduction then (max(eltType), -1) else (min(eltType), -1))`, but it wasn't much harder to do this.

Reply:
In our perf meeting we decided that for now it's probably best to just turn this into an assert so that if it's ever hit you'd know and can address it. Currently it'll work, but will be really slow.